### PR TITLE
Updated the 404 message

### PR DESCRIPTION
**Pull request by hdhalter**
### Description
Updates the 404 text.

### Issues Resolved
N/a

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: 404.md [hunk 1]
```diff
@@ -6,6 +6,15 @@ heading_anchors: false
 nav_exclude: true
 ---
 
-# OpenSearch cannot find that page. 
+## We can't seem to find the page you're looking for. 
+
+Perhaps we moved something around. Here are some of our most popular pages: 
+
+- [Quickstart]({{site.url}}{{site.baseurl}}/quickstart)
```
Comment (kolchfa-aws) [inline 404.md:13]: Add a trailing slash.

**Review by kolchfa-aws — APPROVED** [review]
LGTM with a comment

File: 404.md [hunk 2]
```diff
@@ -6,6 +6,15 @@ heading_anchors: false
...
+## We can't seem to find the page you're looking for. 
+
+Perhaps we moved something around. Here are some of our most popular pages: 
+
+- [Quickstart]({{site.url}}{{site.baseurl}}/quickstart/)
+- [Installing OpenSearch]({{site.url}}{{site.baseurl}}/install-and-configure/install-opensearch/index/)
+- [OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/dashboards/index/)
+- [Query DSL]({{site.url}}{{site.baseurl}}/query-dsl/)
+- [API Reference]({{site.url}}{{site.baseurl}}/api-reference/index/)
+- [OpenSearch documentation (home)]({{site.url}}{{site.baseurl}}/docs/latest/)
```
Comment (vagimeli) [inline 404.md:18]: I suggest moving the doc home page to be the first bullet. 
```suggestion
- [OpenSearch Documentation Homepage]({{site.url}}{{site.baseurl}}/docs/latest/)
```

File: 404.md [hunk 3]
```diff
@@ -6,6 +6,15 @@ heading_anchors: false
 nav_exclude: true
 ---
 
-# OpenSearch cannot find that page. 
+## We can't seem to find the page you're looking for.
```
Comment (vagimeli) [inline 404.md:9]: Suggested text: Oops, this isn't the page you're looking for.

File: 404.md [hunk 4]
```diff
@@ -6,6 +6,15 @@ heading_anchors: false
 nav_exclude: true
 ---
 
-# OpenSearch cannot find that page. 
+## We can't seem to find the page you're looking for. 
+
+Perhaps we moved something around. Here are some of our most popular pages:
```
Comment (vagimeli) [inline 404.md:11]: Suggested text: Maybe our homepage or one of the popular pages listed below can help.

**Review by vagimeli — COMMENTED** [review]
@hdhalter A couple of suggestions to make the copy a little more fun.

**Comment by hdhalter** [comment]
Thanks, Melissa! Do you want to take a final pass? @vagimeli

**Review by vagimeli — APPROVED** [review]

**Comment by hdhalter** [comment]
Before I backport, I want to fix the spacing. Since the line of text is short, it's causing the margins to retract.

**Review by vagimeli — COMMENTED** [review]
LGTM

**Pull request merged**